.github/workflows/publish-winget.yml

```diff
@@ -0,0 +1,36 @@
+###################################################################################################
+###                   THIS IS A REUSABLE WORKFLOW TO PUBLISH SCALA TO WINGET                    ###
+### HOW TO USE:                                                                                 ###
+###   - THE RELEASE WORKFLOW SHOULD CALL THIS WORKFLOW                                          ###
+###   - IT WILL PUBLISH TO WINGET THE MSI                                                       ###
+###                                                                                             ###
+### NOTE:                                                                                       ###
+###   - WE SHOULD KEEP IN SYNC THE https://github.com/dottybot/winget-pkgs REPOSITORY           ###
+###################################################################################################
+
+
+name: Publish Scala to winget
+run-name: Publish Scala ${{ inputs.version }} to winget
+
+on:
+  workflow_call:
+    inputs:
+      version:
+        required: true
+        type: string
+    secrets:
+      DOTTYBOT-TOKEN:
+        required: true
+
+jobs:
+  publish:
+    runs-on: windows-latest
+    steps:
+      - uses: vedantmgoyal9/winget-releaser@b87a066d9e624db1394edcd947f8c4e5a7e30cd7
```

Comment:
let's switch to a version tag once the recent fixes get released.

Reply:
Yep, I've already mentionned this point in https://github.com/scala/scala3/issues/20381#issuecomment-2206430351